utils.py

```diff
@@ -0,0 +1,67 @@
+import requests # pylint: disable=import-error
+from google.cloud import language # pylint: disable=import-error
+from google.cloud.language import enums, types # pylint: disable=import-error
+
+SPANISH_API_URL = "http://sesat.fdi.ucm.es:8080/servicios/rest/sinonimos/json/"
+KEY_PARTS_OF_SPEECH = ["ADJ", "NOUN", "NUM", "VERB"]
+
+def gcloud_syntax_extraction(text):
+    """
+    Calls the gcloud natural langauge api's analyze syntax
+    Attributes:
+        text: the text which we wish to analyze
+    Returns:
+        the response given by the api call
+    """
+    client = language.LanguageServiceClient()
+    # pylint: disable=no-member
+    document = types.Document(
+        content=text,
+        type=enums.Document.Type.PLAIN_TEXT)
+    response = client.analyze_syntax(document=document)
+
+    return response
+
+def from_gcloud_remove_extra(gcloud_response):
+    """
+    Removes extra information from the gcloud response. This includes everything that is not
+    the language, the words found, the lemma version of said words, the part of speech for said word
+    and any word that is not part of the key parts of speech constants.
+    Attributes:
+        gcloud_response: the response given by the natural language api
+    Returns:
+        An object with the language, the keywords as defined by the design doc,
+        the lemma and part of speech for said keywords
+    """
+    tokens_shortened = []
+    for token in gcloud_response.tokens:
```

Comment:
I haven't looked into the API, but is there any way for you to model the response format on the same query? Like trying to only fetch the information you want rather than downloading all the payload and curating it in here.

Can you pass any sort of params or so to define how the response should look like?

I am asking because you've been diving deep into this documentation, so I'm pretty sure you have better insights.

I think that if you find anything of the sort, you could reduce CPU cycles and overall computation and iterations you have to perform to extract what matters to you from the response. If it's not possible, then I guess this is the way to go, but it'd be good to know if there's anything available.

Reply:
All examples of this api call I've seen seem to indicate that there's no way to filter down the response